tests/blendernc_performance.test.py

```diff
@@ -0,0 +1,109 @@
+import os
+import sys
+import time
+import unittest
+import warnings
+
+import bpy
+import dask.array.core as darc
+import numpy as np
+
+warnings.simplefilter(action="ignore", category=darc.PerformanceWarning)
+
+
+def capture_render_log(func):
+    def wrapper(*args, **kwargs):
+        logfile = "blender_render.log"
+        open(logfile, "a").close()
+        old = os.dup(1)
+        sys.stdout.flush()
+        os.close(1)
+        os.open(logfile, os.O_WRONLY)
+        func(*args, **kwargs)
+        os.close(1)
+        os.dup(old)
+
+    return wrapper
+
+
+@capture_render_log
+def render_image(file, var, res):
+
+    bpy.context.scene.blendernc_datacube_vars = var
+
+
+class Test_format_import(unittest.TestCase):
+    def test_performance_ssh(self):
+        file = os.path.abspath("./dataset/ssh_1995-01.nc")
+        var = "adt"
+        res = 100
+        samples = 100
+        times = np.zeros(samples)
+        for n in range(samples):
+            bpy.ops.wm.read_homefile()
+            bpy.context.scene.blendernc_file = file
+            bpy.context.scene.blendernc_resolution = res
+            tic = time.time()
+            render_image(file, var, res)
+            toc = time.time()
+            times[n] = toc - tic
+
+        print("[", end="")
+        for tt in times:
+            print(str(tt) + ",", end="")
+        print("]", end="\n")
+        # Frames must be more than 2
+        self.assertGreater(1 / np.mean(times), 2)
+
+    def test_performance_t2m(self):
+        file = os.path.abspath("./dataset/ECMWF_data.grib")
+        var = "t2m"
+        res = 100
+        samples = 100
+        times = np.zeros(samples)
+        for n in range(samples):
+            bpy.ops.wm.read_homefile()
+            bpy.context.scene.blendernc_file = file
+            bpy.context.scene.blendernc_resolution = res
+            tic = time.time()
+            render_image(file, var, res)
+            toc = time.time()
+            times[n] = toc - tic
+
+        print("[", end="")
+        for tt in times:
+            print(str(tt) + ",", end="")
+        print("]", end="\n")
+        # Frames must be more than 5
+        self.assertGreater(1 / np.mean(times), 5)
+
+
+suite = unittest.defaultTestLoader.loadTestsFromTestCase(Test_format_import)
+test = unittest.TextTestRunner().run(suite)
+
+ret = not test.wasSuccessful()
+sys.exit(ret)
+
+
+# # To plot
```

Comment:
I love the plots. I believe they should be created as part of this test and saved as GitHub artifacts. https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts